### ISSUE TITLE
fix(spherical): GFD solvers rebuild `SphericalDerivative2D` (and scipy tables) on every call and cannot be jitted

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#98]
Part of #81.

## Summary

`SphericalVorticityInversionSolver.solve`, `SphericalDivergenceInversionSolver.solve` and `SphericalHelmholtzDecomposition.decompose` construct a `SphericalPoissonSolver` and a `SphericalDerivative2D` inside the method body. `SphericalDerivative2D.__init__` builds a fresh `SphericalGrid1D` (scipy `lpmv` + Gauss–Legendre nodes), and `SphericalDerivative1D.__init__` does `np.array(grid._nodes)` (`operators.py:130`). Two consequences:

1. **Impossible to `jax.jit`.** Under any trace, `grid._nodes` is a tracer and `np.array(...)` raises `TracerArrayConversionError`, even when the solver is closed over and even under `eqx.filter_jit`.
2. **Eager cost dominated by table construction.** 39–250 ms per call at `Nx = 32, Ny = 16` for ≤ 0.3 ms of arithmetic.

`vmap` and `grad` happen to work because they do not stage the construction.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from spectraldiffx import SphericalGrid2D, SphericalVorticityInversionSolver
g = SphericalGrid2D.from_N_L(Nx=32, Ny=16, Lx=2*np.pi, Ly=np.pi)
s = SphericalVorticityInversionSolver(g)
zeta = jnp.sin(g.y)[:, None] * jnp.cos(g.x)[None, :]
jax.jit(lambda z: s.solve(z))(zeta)      # TracerArrayConversionError at operators.py:130
```

Same for `SphericalHelmholtzDecomposition.decompose`, `eqx.filter_jit(lambda s, z: s.solve(z))(s, zeta)`, and `jax.jit(lambda x: SphericalDerivative2D(g).gradient(x))`.

## Root cause

- `spectraldiffx/_src/spherical/solvers.py:274-277, 320-322, 410-415`: operators built per call.
- `spectraldiffx/_src/spherical/operators.py:127-134, 296-305`: `SphericalGrid1D` / numpy conversion inside `__init__`.

## Proposed fix

1. Build the derivative and Poisson operators once in each solver's `__init__` and store them as fields (the same way `deriv_theta` is already a field on `SphericalDerivative2D`).
2. In `SphericalDerivative1D.__init__`, compute nodes from `_gauss_legendre_nodes_weights(grid.N)` in numpy rather than converting `grid._nodes` back from JAX.
3. Longer term (with the stable-recurrence issue in this epic): build the tables in JAX with `lax.scan`, so construction itself is traceable and `eqx.filter_jit` over a freshly constructed operator works.

Once the spectral θ-derivative issue lands, `SphericalDerivative1D` is no longer needed on the 2D path at all, which removes the scipy dependency from `SphericalDerivative2D` entirely.

## Acceptance criteria

- [ ] `jax.jit(lambda z: solver.solve(z))` works for all three GFD solvers; a test asserts this and compares to the eager result to 1e-12.
- [ ] A test counts `SphericalDerivative2D` constructions (monkeypatch) across three `solve` calls: exactly zero after `__init__`.
- [ ] Eager `solve` at `Nx = 32, Ny = 16` under 2 ms.
